components/schedule-editor/src/ScheduleEditor.svelte

```diff
@@ -519,676 +532,705 @@
+                        <input
+                          type="text"
+                          aria-label="Description"
+                          bind:value={newCustomField.description} />
+                      </label>
+                    </div>
+                    <div class="input-field">
+                      <label class="select">
+                        <div>Type</div>
+                        <select
+                          bind:value={newCustomField.type}
+                          aria-label="Input Type">
+                          <option value="text">Text</option>
+                          <option value="checkbox">Checkbox</option>
+                        </select>
+                      </label>
+                    </div>
+                    <div class="input-field">
+                      <label class="checkbox">
+                        <input
+                          type="checkbox"
+                          aria-label="Required Field?"
+                          bind:checked={newCustomField.required} />
+                        <span>Required</span>
+                      </label>
+                    </div>
+                    <div class="footer">
                       <button
-                        class="drag"
-                        on:mousedown={(event) => {
-                          handleCustomFieldDrag(event, field, i);
+                        class="cancel"
+                        on:click={() => {
+                          newCustomField = { ...emptyCustomField };
+                          isAddingCustomField = false;
                         }}>
-                        <span class="sr-only">Reorder custom field</span>
-                        <DragIcon />
+                        <span>Cancel</span>
                       </button>
-                    </td>
-                    <td
-                      class:multi-line={field.description}
-                      class={"title-and-description"}>
-                      <span>{field.title ?? "—"}</span>
-                      {#if field.description}
-                        <span class="description">{field.description}</span>
-                      {/if}
-                    </td>
-                    <td class="type">{field.type ?? "—"}</td>
-                    <td class="required">
-                      {#if field.required}
-                        <CheckmarkIcon />
-                      {/if}
-                    </td>
-                    <td>
                       <button
-                        class="edit"
-                        aria-label="Edit"
-                        on:click={() => editCustomField(field)}>
-                        <PencilIcon />
-                        <span>Edit</span>
+                        class="save"
+                        disabled={newCustomField.title?.length === 0 ||
+                          !newCustomField.type}
+                        on:click={() => addNewField()}>
```

Comment:
Can it be like this?
```suggestion
                        on:click={addNewField}>
```

Reply:
Nope; the way you suggest would automatically run the function on component init. Passing event handlers as functions is the correct way to handle them in Svelte.

Reply:
`() => addNewField()` and `addNewField` are functionally equivalent, except it will pass an event to the function. But `addNewField` doesn't accept any params.
If `addNewField` is called on component init, then `() => addNewField()` will be called as well. Unless this is Svelte's way.